app/models/energy/ScheduleRuleset.py

```diff
@@ -0,0 +1,113 @@
+"""Schedule Ruleset Schema"""
+from pydantic import BaseModel, Schema, validator, ValidationError, UrlStr, ConstrainedStr
+from typing import List, Union
+from enum import Enum
+from uuid import UUID, uuid4
+from datetime import datetime
+from app.models.energy.ScheduleBase import ScheduleContinuous, ScheduleDiscrete, ScheduleNumericType, ScheduleUnitType, DateTime, Date, Time
+
+
+class ScheduleTypeLimits (BaseModel):
+    """Specifies the data types and limits for values contained in schedules."""
+
+    type: Enum('ScheduleTypeLimits', {'type': 'ScheduleTypeLimits'})
+
+    name: str = Schema(
+        ...,
+        regex=r'^[\s.A-Za-z0-9_-]*$',
+    )
+
+    lower_limit_value: float = Schema(
+        0,
+        description='Lower limit for the schedule type is entered.'
+    )
+
+    upper_limit_value: float = Schema(
+        1,
+        description='Upper limit for the schedule type is entered.'
+    )
+
+    numeric_type: ScheduleNumericType
+
+    unit_type: ScheduleUnitType
+
+
+class DayValue(BaseModel):
+    """Values for daily schedule"""
+
+    time: Time
+
+    value_until_time: float
+
+
+class ScheduleDay(BaseModel):
+    """Used to describe the daily schedule for a single simulation day."""
+    type: Enum('ScheduleDay', {'type': 'ScheduleDay'})
+
+    name: str = Schema(
+        ...,
+        regex=r'^[\s.A-Za-z0-9_-]*$',
+    )
+
+    interpolate_to_timestep: bool
+
+    day_values: List[DayValue]
+
+
+class ScheduleRule(BaseModel):
+    """A set of rules assigned to schedule ruleset for specific periods of time and for
+  particular days of the week according to a priority sequence."""
+
+    type: Enum('ScheduleRule', {'type': 'ScheduleRule'})
+
+    name: str = Schema(
+        ...,
+        regex=r'^[\s.A-Za-z0-9_-]*$'
+    )
+
+    schedule_day: ScheduleDay
+
+    apply_sunday: bool
```

Comment:
I am a little confused about what happens if I apply two schedule rules for sunday (or any other day of the week). Do we need a validator to make sure that we don't have two rules occupying the same day?

Reply:
@chriswmackey In that case the schedule rule earlier in the list of schedule rules would be given priority.

Reply:
That is the OpenStudio rule priority logic.  It can be a bit confusing, we have our own schedule GUI to show which rule is active on each day.  Not sure if you plan on implementing any type of calendar viewer in your UI?

Reply:
Good to know, Dan.  If we make any calendar GUI, it is going to be developed pretty far down the line.